Source/WebCore/loader/cache/CachedResource.cpp

```diff
@@ -779,9 +779,9 @@ void CachedResource::switchClientsToRevalidatedResource()
     ASSERT(!m_handleCount);
     m_handlesToRevalidate.clear();
 
-    Vector<CachedResourceClient*> clientsToMove;
-    for (auto& entry : m_clients) {
-        CachedResourceClient* client = entry.key;
+    Vector<WeakPtr<CachedResourceClient>> clientsToMove;
```

Comment:
If we're certain that these pointers won't be deleted during this operation, then I think the best option is Vector<CachedResourceClient&>.

But I suppose that removeClient() and didAddClient() might have arbitrary side effects? If so, I guess we need to add some null checks when we call removeClient() and addClientToSet().

Reply:
Added null checks.